theano/compile/builders.py

```diff
@@ -38,11 +40,12 @@ class OpFromGraph(gof.Op):
 
         ``False`` : will use a pre-compiled function inside.
 
-    grad_overrides : single or list of {None, undef, OpFromGraph, callable}, optional
+    grad_overrides : single or list of {0, None, Ellipsis, OpFromGraph, callable}, optional
         Defaults to ``None``.
 
-        ``None`` : Do not override gradient
-        theano.utils.undef : No gradient will be used (zero)
+        ``None`` : No value, gives NullType()
+        ``0`` : zero value, gives DisconnectedType()
+        ``...`` : Do not override, use default grad() result
```

Comment:
@lamblin What do you think of this? I'm not convinced. I would think that using None for do not override made more sence. None mean do thing in subtensor semantic. But None is also close in writing to Null.

Why not ask the user to give NullType and DisconnectedType instance directly to be more explicit? We removed None in grad everywhere else I think to be more explicit. Also, using 0 for Disconected could in theory mix cases where we are able to infer the value of grad to be always 0 during theano.grad() and not disconnected. We will probably never do that, but I'm pretty sure in some corner case the graph would allow us to do so.